packages/language-server/src/hover.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  TextDocumentPositionParams,
+  TextDocument,
+  Hover,
+  MarkupContent,
+  MarkupKind,
+} from "vscode-languageserver";
+import { parse, DocumentCstNode } from "@xml-tools/parser";
+import { buildAst } from "@xml-tools/ast";
+import { getAstNodeInPosition } from "@xml-tools/ast-position";
+import {
+  UI5SemanticModel,
+  BaseUI5Node,
+  UI5Field,
+  UI5Prop,
+  UI5Aggregation,
+  UI5Association,
+} from "@ui5-language-assistant/semantic-model-types";
+import { findUI5HoverNodeAtOffset } from "@ui5-language-assistant/xml-views-tooltip";
+import { getNodeDocumentation } from "./documentation";
+import {
+  isRootSymbol,
+  ui5NodeToFQN,
+  typeToString,
+} from "@ui5-language-assistant/logic-utils";
+
+export function getHoverResponse(
+  model: UI5SemanticModel,
+  textDocumentPosition: TextDocumentPositionParams,
+  document: TextDocument
+): Hover | undefined {
+  const documentText = document.getText();
+  const { cst, tokenVector } = parse(documentText);
+  const ast = buildAst(cst as DocumentCstNode, tokenVector);
+  const offset = document.offsetAt(textDocumentPosition.position);
+  const astPosition = getAstNodeInPosition(ast, offset);
+  if (astPosition !== undefined) {
+    const ui5Node = findUI5HoverNodeAtOffset(astPosition, model);
```

Comment:
**Not for now** but maybe we should rename `findUI5HoverNodeAtOffset` because to `findUI5HoverNodeAtPosition`